src/groups/mqb/mqbnet/mqbnet_tcpsessionfactory.cpp

```diff
@@ -1269,20 +1395,20 @@ void TCPSessionFactory::stop()
     BALL_LOG_INFO << "Stopped TCPSessionFactory '" << d_config.name() << "'";
 }
 
-int TCPSessionFactory::listen(int port, const ResultCallback& resultCallback)
+int TCPSessionFactory::listen(const mqbcfg::TcpInterfaceListener& listener,
+                              const ResultCallback& resultCallback)
 {
-    // Supporting only one 'listen' call.
-    BSLS_ASSERT_SAFE(!d_isListening);
-    BSLS_ASSERT_SAFE(!d_listenContext_mp);
+    const int port = listener.port();
+
+    BSLS_ASSERT_SAFE(d_listenContexts.find(port) == d_listenContexts.cend());
 
     // Maintain ownership of 'OperationContext' instead of passing it to
     // 'ChannelFactory::listen' because it may delete the context
     // (on stopListening) while operation (readCallback / negotiation) is in
     // progress.
-    OperationContext* context = new (*d_allocator_p) OperationContext();
-    d_listenContext_mp.load(context, d_allocator_p);
-    bsl::shared_ptr<OperationContext> contextSp;
-    contextSp.reset(context, bslstl::SharedPtrNilDeleter());
```

Comment:
Yeah, this insisting on the `ManagedPtr` looks strange

Reply:
Yeah I'm not in love with this pattern of trading around `ManagedPtr`s as references wrapped in a `shared_ptr`, either way now there's a reason to make it shared so this doesn't even work anymore 😄